alws/crud/products.py

```diff
@@ -338,18 +364,95 @@ async def modify_product(
     perform_product_modification.send(db_build.id, db_product.id, modification)
 
 
-async def get_repo_product(session: AsyncSession, repository: str) -> Optional[Product]:
+async def add_platform_to_product(
+    session: AsyncSession,
+    product_id: int,
+    platforms: List[Platform],
+):
+    pulp_client = PulpClient(
+        settings.pulp_host,
+        settings.pulp_user,
+        settings.pulp_password,
+    )
+    db_product = await get_products(session, product_id=product_id)
+    db_platforms = (
+        (
+            await session.execute(
+                select(models.Platform).where(
+                    models.Platform.name.in_(
+                        [platform.name for platform in platforms]
+                    ),
+                ),
+            )
+        )
+        .scalars()
+        .all()
+    )
+    db_product.platforms.extend(db_platforms)
+    items_to_insert = []
+    repo_tasks = []
+    for platform in db_platforms:
+        platform_name = platform.name.lower()
+        repo_tasks.extend((
+            create_product_repo(
+                pulp_client,
+                db_product.name,
+                db_product.owner.username,
+                platform_name,
+                arch,
+                is_debug,
+            )
+            for arch in platform.arch_list
+            for is_debug in (True, False)
+        ))
+        repo_tasks.append(
+            create_product_repo(
+                pulp_client,
+                db_product.name,
+                db_product.owner.username,
+                platform_name,
+                'src',
+                False,
+            )
+        )
+    task_results = await asyncio.gather(*repo_tasks)
+
+    for repo_name, repo_url, arch, pulp_href, is_debug in task_results:
+        repo = models.Repository(
+            name=repo_name,
+            url=repo_url,
+            arch=arch,
+            pulp_href=pulp_href,
+            type=arch,
+            debug=is_debug,
+            production=True,
+        )
+        db_product.repositories.append(repo)
+        items_to_insert.append(repo)
```

Comment:
Most logic of the new function repeats the same logic as in the `create_product` function, please move common logic to a separate function and reuse it